public/plugin.ts

```diff
@@ -35,7 +35,17 @@ export class OpendistroSecurityPlugin
   constructor(private readonly initializerContext: PluginInitializerContext) {}
 
   public async setup(core: CoreSetup): Promise<OpendistroSecurityPluginSetup> {
-    const permissions = await core.http.get(API_ENDPOINT_PERMISSIONS_INFO);
+    let permissions;
+
+    try {
+      permissions = await core.http.get(API_ENDPOINT_PERMISSIONS_INFO);
+    } catch (e) {
+      if (e.body.statusCode === 401) {
+        // ignore and continue to login page.
+      } else {
+        throw e;
+      }
+    }
 
     if (permissions?.data?.has_api_access) {
```

Comment:
Suggest extract to separate function so easier to add unit test later.

Reply:
ok